tests/unit/algos/list/delimiterMaster.spec.js

```diff
@@ -46,7 +48,91 @@ function getListingKey(key, vFormat) {
     return assert.fail(`bad vFormat ${vFormat}`);
 }
 
-['v0', 'v1'].forEach(vFormat => {
+const MAX_STREAK_LENGTH = 100;
+const params = {};
+let streakLength = 0;
+let gteparams = null;
+
+/**
+ * Simplified version of logic found in Metadata's RepdServer. When MAX_STREAK_LENGTH (typically 100)
+ * is reached, we attempt to skip to the next listing range as a performance optimization. 
+ * @param {Integer} filteringResult - 0, 1, -1 indicating if we can skip to the next character range.
+ * @param {String} skippingRange - lower bound that the listing can begin from.
+ * @returns 
+ */
+function handleStreak(filteringResult, skippingRange) {
+    if (filteringResult < 0) {
+        // readStream would be destroyed. In this mock, we just continue.
+    } else if (filteringResult === 0 && skippingRange) {
+        // check if MAX_STREAK_LENGTH consecutive keys have been
+        // skipped
+        if (++streakLength === MAX_STREAK_LENGTH) {
+            if (Array.isArray(skippingRange)) {
+                // With synchronized listing, the listing
+                // algo backend skipping() function must
+                // return as many skip keys as listing
+                // param sets.
+                for (let i = 0; i < skippingRange.length; ++i) {
+                    // eslint-disable-next-line no-param-reassign
+                    params[i].gte = inc(skippingRange[i]);
+                }
+            } else {
+                // eslint-disable-next-line no-param-reassign
+                params.gte = inc(skippingRange);
+            }
+            if (gteparams && gteparams === params.gte) {
+                streakLength = 1;
+            } else {
+                // stop listing this key range
+                // eslint-disable-next-line no-param-reassign
+                gteparams = params.gte;
+            }
+        }
+    } else {
+        streakLength = 1;
+    }
+    return undefined;
+}
+
+/**
+ * Generate a random number of versioned keys.
+ * @param {Integer} maxKeys - how many versioned keys to generate
+ * @yields {Object} - { key, isDeleteMarker }
+ */
+function *generateVersionedKeys(masterKey, numKeys) {
+    let [idx, key] = [0, masterKey];
+    while (idx < numKeys) {
+        key = `${masterKey}${VID_SEP}${zpad(idx)}`
+        yield {key, isDeleteMarker: chance.bool()}
+        idx++
+    }
```

Comment:
I would just use a for loop here
```suggestion
    for (let i = 0; i < numKeys; i++) {
        yield {
            key: `${masterKey}${VID_SEP}${zpad(i)}`, 
            isDeleteMarker: chance.bool(),
        }
    }
```
For key you could also do `[masterKey, VID_SEP, zpad(i)].join('')` or just `+`, I never know what is better :smirk: